### PR TITLE
Refactor website build script

### PR DESCRIPTION
**Pull request by fisker**
## Description

- Reuse existing utils
- Replace `shelljs`
- Rename `build-docs` -> `build:website`, `build-docs.mjs` -> `build-website.mjs`

## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: scripts/utils/index.mjs [hunk 1]
```diff
@@ -30,5 +30,15 @@ async function writeFile(file, content) {
 const PROJECT_ROOT = url.fileURLToPath(new URL("../../", import.meta.url));
 const CACHE_DIR = path.join(PROJECT_ROOT, ".cache");
```
Comment (sosukesuzuki) [inline scripts/utils/index.mjs:31]:
```suggestion
const BUILD_CACHE_DIR = path.join(PROJECT_ROOT, ".cache");
```

File: scripts/build-website.mjs [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  try {
+    await runYarn("build", ["--playground"], { cwd: PROJECT_ROOT });
+  } finally {
+    // restore
+    await writeFile(packageJsonFile, content);
+  }
+}
+
+async function buildPlaygroundFiles() {
+  const files = globby.sync(["standalone.js", "parser-*.js"], {
```
Comment (sosukesuzuki) [inline scripts/build-website.mjs:51]: `buildPlaygroundFiles` is an async function, so why use `globby.sync`?

File: scripts/build-website.mjs [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  for (const fileName of files) {
+    const file = path.join(PRETTIER_PATH, fileName);
+    await copyFile(file, path.join(PLAYGROUND_PRETTIER_DIR, fileName));
+
+    if (fileName === "standalone.js") {
+      continue;
+    }
+
+    const plugin = require(file);
+    const property = fileName.replace(/\.js$/, "").split("-")[1];
```
Comment (sosukesuzuki) [inline scripts/build-website.mjs:64]: Please add comment on example  of specific `property` value.

File: netlify.toml [hunk 4]
```diff
@@ -1,3 +1,6 @@
+[build]
+  command = "yarn build:website"
+
 [build.environment]
-  NODE_VERSION = "14"
+  NODE_VERSION = "16"
```
Comment (sosukesuzuki) [inline netlify.toml:5]: Just a question, what feature of Node.js 16 we are using?

Comment (sosukesuzuki) [reply netlify.toml:5]: Also, I think we should mention in CONTRIBUTING.md that we need Node.js 16 for building our project.

**Review by sosukesuzuki — COMMENTED** [review]
Good refactoring! Please check my comments

File: netlify.toml (hunk 4 above)
Comment (fisker) [reply netlify.toml:5]: It's not required, just want to use the newer version.

Comment (sosukesuzuki) [reply netlify.toml:5]: I prefer using Node.js 14 that is current LTS version.

**Review by sosukesuzuki — APPROVED** [review]

**Comment by fisker** [comment]
I also changed the build image from `Ubuntu Xenial 16.04` to `Ubuntu Focal 20.04 (default)` in the netlify web ui https://app.netlify.com/sites/prettier/settings/deploys

**Comment by fisker** [comment]
> I also changed the build image from Ubuntu Xenial 16.04 to Ubuntu Focal 20.04 (default)

It breaks the website build, I switched back.

**Pull request merged**